spec/marten/db/query/set_spec.cr

```diff
@@ -1298,6 +1298,245 @@ describe Marten::DB::Query::Set do
       qset.to_a.sort_by(&.pk!.to_s).should eq [tag_2, tag_3].sort_by(&.pk!.to_s)
     end
 
+    it "filters records using a raw SQL equality condition", tags: "raw" do
+      Tag.create!(name: "ruby", is_active: true)
+      tag_2 = Tag.create!(name: "crystal", is_active: true)
+      Tag.create!(name: "coding", is_active: true)
+      Tag.create!(name: "programming", is_active: true)
+
+      qset = Marten::DB::Query::Set(Tag).new.filter("name='crystal'")
+
+      qset.to_a.should eq [tag_2]
+    end
+
+    it "does not filter records using a empty raw SQL equality condition", tags: "raw" do
+      tag_1 = Tag.create!(name: "ruby", is_active: true)
+      tag_2 = Tag.create!(name: "crystal", is_active: true)
+      tag_3 = Tag.create!(name: "coding", is_active: true)
+      tag_4 = Tag.create!(name: "programming", is_active: true)
+
+      qset = Marten::DB::Query::Set(Tag).new.filter("")
+
+      qset.to_a.should eq [tag_1, tag_2, tag_3, tag_4]
+    end
+
+    it "filters records using a raw SQL condition with one named parameters", tags: "raw" do
+      Tag.create!(name: "ruby", is_active: true)
+      tag_2 = Tag.create!(name: "crystal", is_active: true)
+      Tag.create!(name: "coding", is_active: true)
+      Tag.create!(name: "programming", is_active: true)
+
+      qset = Marten::DB::Query::Set(Tag).new.filter("name=:name", name: "crystal")
+
+      qset.to_a.should eq [tag_2]
+    end
+
+    it "raises an error when filtering with a misspelled column in a raw SQL condition", tags: "raw" do
+      Tag.create!(name: "crystal", is_active: true)
+
+      expect_raises(Exception) do
```

Comment:
We could, but I'm not sure it's useful here. All database types throw their own exception. I didn't see a common ancestor of these database exceptions, so I chose this generic exception type instead of checking each individual database exception.

What do you think?

Reply:
We could define specific blocks for every supported backend and verify that the most precise exception is raised (by wrapping `#expect_raises` calls within `#for_mysql` / `#for_postgresql` / `#for_sqlite` blocks). That way, other types of exceptions wouldn't be "caught" if there is an issue with the underlying code, but that's very unlikely anyway. Let's keep it that way!